modules/core/04-channel/keeper/packet.go

```diff
@@ -342,14 +342,15 @@ func (k Keeper) WriteAcknowledgement(
 		return types.ErrAcknowledgementExists
 	}
 
-	if len(acknowledgement) == 0 {
+	bz := acknowledgement.Acknowledgement()
+	if len(bz) == 0 {
```

Comment:
```suggestion
	if acknowledgement == nil || len(acknowledgement.Acknowledgement()) == 0 {
```

Reply:
Added the check as a separate if statement. Just my personal preference.